lib/exercises.rb

```diff
@@ -2,19 +2,67 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  results = []
+  return results if strings.empty?
+
+  hash_map = {}
+  set_no = 1  
+
+  strings.each do |word|
+    letters = word.split('').sort!
+
+    if hash_map.empty?
+      hash_map[set_no] = letters
+      set_no += 1
+    elsif !hash_map.has_value?(letters)
+      hash_map[set_no] = letters
+      set_no += 1
+    end 
+  end 
+
+  results = []
+
+  hash_map.keys.each do |sub_arr|
+    sub_arr = []
+    results << sub_arr
+  end 
+
+  strings.each do |word|
+    letters = word.split('').sort!
+
+    if hash_map.has_value?(letters)
+      sub_group = hash_map.key(letters)
```

Comment:
Using `.key` searches all the values and finds the 1st key that has the given value and is O(n), since you're doing this n times...